File: python/binary_search.py
import math

a = [2, 4, 5, 8]


def binary_search(array, value):
    lo = 0
    hi = len(array)

    while lo < hi:
        m = math.floor(lo + (hi - lo) / 2)

        print(m)

        if array[m] == value:
            return True
        elif array[m] > value:
            hi = m
        elif array[m] < value:
            lo = m + 1

    return False


print(binary_search(a, 5))
print(binary_search(a, 6))

File: go/binary_search/binary_search.go
package algorithm

import (
	"fmt"
	"math"
)

func binary_search(array [4]int, v int) bool {
	lo := 0
	hi := len(array)

	for lo < hi {
		m := int(math.Floor(float64(lo + (hi-lo)/2)))

		if array[m] == v {
			return true
		} else if array[m] > v {
			hi = m
		} else {
			lo = m + 1
		}
	}

	return false
}

func main() {
	a := [4]int{2, 4, 5, 8}

	fmt.Println(binary_search(a, 5))
	fmt.Println(binary_search(a, 6))
}
